src/vcr_cleaner/__init__.py

```diff
@@ -20,20 +24,107 @@ def serialize(self, cassette: dict):
     def deserialize(cassette: str):
         return yamlserializer.deserialize(cassette)
 
-    def register_cleaner(self, function: Callable, uri=None):
-        if uri:
-            @clean_if(uri=uri)
-            def decorated(*args, **kwargs):
-                function(*args, **kwargs)
-        self.cleaners.append(decorated if uri else function)
+    def register_cleaner(self, function: Callable):
+        '''Apply this cleaner function to all cassette contents.'''
+        self.cleaners.append(function)
+
+    def register_cleaner_if_host_startswith(
+        self, function: Callable, hostname: str
+    ):
+        '''
+        Apply registered cleaner only if URI hostname before the first `/`
+        matches the given text.
+
+        'example.com' will match 'https://example.com/robots.txt'
+        'example.com' will match 'https://example.com/api/v1/users'
+        '''
+
+        @_apply_if_uri_startswith(uri_starts=hostname)
+        def decorated(*args, **kwargs):
+            function(*args, **kwargs)
+        self.register_cleaner(decorated)
+
+    def register_cleaner_if_path_endswith(self, function: Callable, path: str):
+        '''
+        Apply registered cleaner only if URI path after the first `/`
+        matches the given text.
+        '''
+
+        @_apply_if_uri_endswith(uri_ends=path)
+        def decorated(*args, **kwargs):
+            function(*args, **kwargs)
+        self.register_cleaner(decorated)
+
+    def register_cleaner_if_uri_contains(
+        self,
+        function: Callable,
+        uri_contains: str
+    ):
+        '''Apply registered cleaner only if URI contains the given text.'''
+
+        @_apply_if_uri_contains(uri_contains=uri_contains)
+        def decorated(*args, **kwargs):
+            function(*args, **kwargs)
+        self.register_cleaner(decorated)
+
+
+def _apply_if_uri_contains(uri_contains: str):
+    '''Decorates a cleaner method to make it apply to any cassette interactions
+    where the URI contains the text passed to `uri_contains`.
+
+    Typically applied by the matching `register_cleaner_if...` function above.
+    '''
 
+    def decorator(func: Callable):
+        @functools.wraps(func)
+        def wrapper(request: dict, response: dict):
+            if uri_contains not in request['uri']:
+                return
+            func(request, response)
```

Comment:
I think the code would be more straight forward, and ever so slightly faster if we drop the use of `not`. I suggest not for performance but simplicity and readability:

```suggestion
            if uri_contains in request['uri']:
                func(request, response)
```